File: sdks/sdkwork-mail-sdk/sdkwork-mail-sdk-swift/Sources/MailSdk/mailStandardContract.swift
public enum MailStandardContract {
    public static let symbol = "MailStandardContract"
}

public protocol MailProviderDriver {
    var providerKey: String { get }
}

public protocol MailDriverManager {
    func resolveDriver(providerKey: String)
}

public protocol MailDataSource {
    func createClient() async throws
}

public protocol MailClient {
    func connectTransport() async throws
    func authenticateTransport() async throws
    func disconnectTransport() async throws
    func sendMail() async throws
    func probeMailbox() async throws
    func syncMailbox() async throws
    func healthCheck() async throws
    func unwrap() -> Any?
}

public protocol MailRuntimeController {
    func connectTransport() async throws
    func authenticateTransport() async throws
    func disconnectTransport() async throws
    func sendMail() async throws
    func probeMailbox() async throws
    func syncMailbox() async throws
    func healthCheck() async throws
}

File: sdks/sdkwork-mail-sdk/sdkwork-mail-sdk-csharp/src/SDKWork.Mail.Sdk/mailStandardContract.cs
namespace Sdkwork.Mail.Sdk;

public static class MailStandardContract
{
    public const string Symbol = "MailStandardContract";
}

public interface MailProviderDriver<TNativeClient>
{
    string ProviderKey { get; }

    MailClient<TNativeClient> CreateClient();
}

public interface MailDriverManager<TNativeClient>
{
    MailProviderDriver<TNativeClient> ResolveDriver(string providerKey);
}

public interface MailDataSource<TNativeClient>
{
    MailClient<TNativeClient> CreateClient();
}

public interface MailClient<TNativeClient>
{
    void ConnectTransport();

    void AuthenticateTransport();

    void DisconnectTransport();

    void SendMail();

    void ProbeMailbox();

    void SyncMailbox();

    void HealthCheck();

    TNativeClient? Unwrap();
}

public interface MailRuntimeController<TNativeClient>
{
    void ConnectTransport();

    void AuthenticateTransport();

    void DisconnectTransport();

    void SendMail();

    void ProbeMailbox();

    void SyncMailbox();

    void HealthCheck();
}
